uvicorn/main.py

```diff
@@ -256,28 +256,34 @@ def __init__(self, config):
         self.should_exit = False
         self.force_exit = False
 
-    def run(self):
+    def main(self):
+        self.config.setup_event_loop()
+
+        self.loop.run_until_complete(self.run())
+        self.loop.stop()
+
+    async def run(self):
         process_id = os.getpid()
 
         config = self.config
         if not config.loaded:
             config.load()
 
-        config.setup_event_loop()
-
-        self.loop = asyncio.get_event_loop()
         self.logger = config.logger_instance
         self.lifespan = config.lifespan_class(config)
 
         self.install_signal_handlers()
 
         self.logger.info("Started server process [{}]".format(process_id))
-        self.loop.run_until_complete(self.startup())
-        self.loop.run_until_complete(self.main_loop())
-        self.loop.run_until_complete(self.shutdown())
-        self.loop.stop()
+        await self.startup()
+        await self.main_loop()
+        await self.shutdown()
         self.logger.info("Finished server process [{}]".format(process_id))
 
+    @property
+    def loop(self):
+        return asyncio.get_event_loop()
```

Comment:
I think we should probably drop this, and just use `loop = asyncio.get_event_loop()` inside any methods that need it.